extensions/cornerstone/src/initMeasurementService.js

```diff
@@ -252,7 +253,24 @@ const connectMeasurementServiceToTools = (
         SOPInstanceUID
       );
 
-      const imageId = dataSource.getImageIdsForInstance({ instance });
+      let imageId;
+      let frameNumber = 1;
+
+      if (measurement.metadata && measurement.metadata.referencedImageId) {
+        imageId = measurement.metadata.referencedImageId;
+        frameNumber = getSOPInstanceAttributes(
+          measurement.metadata.referencedImageId
+        ).frameNumber;
+      } else if (measurement.data && measurement.data.frameNumber) {
```

Comment:
`frameNumber` does not get stored in the `data` ever, why are you checking it here?

Reply:
fixed in 9348a17652a77155bf7cd129fc17f2b4993dc785